src/Forms/GridField/GridField.php

```diff
@@ -630,6 +631,21 @@ public function FieldHolder($properties = [])
             }
         }
 
+        // Set to read-only if sudo mode is required for the DataObject being managed
+        if (is_a($list, DataList::class)) {
```

Comment:
Scenario is a has_many relation GridField e.g. Page has_many Member, this is the top level gridfield, not the edit form